### PR TITLE
Investor invite accept form

### PR DESCRIPTION
**Pull request by alexmatson-da**

### CODE REVIEW

File: ui/src/components/Investor/Investor.tsx [hunk 1]
```diff
@@ -25,21 +34,41 @@ type Props = {
     onLogout: () => void;
 }
 
+const useWellKnown = () => {
```
Comment (Sarahb-da) [inline ui/src/components/Investor/Investor.tsx:37]: can we make this hook common?

File: ui/src/components/Issuer/IssueAsset.tsx [hunk 2]
```diff
@@ -57,19 +58,16 @@ const IssueAsset = () => {
...
+
+            <FormToggle
+                defaultChecked
+                className='issue-asset-form-item'
+                onLabel='Public'
+                onInfo='All parties will be aware of this token.'
+                offLabel='Private'
+                offInfo='Only a set of parties will be aware of this token.'
+                onClick={val => setIsPublic(val)}/>
+
```
Comment (Sarahb-da) [inline ui/src/components/Issuer/IssueAsset.tsx:70]: ty

File: ui/src/components/Investor/Investor.tsx [hunk 3]
```diff
@@ -66,4 +95,41 @@ const Investor: React.FC<Props> = ({ onLogout }) => {
...
 
+const InvestorSetup: React.FC = () => {
+    const [ isPublic, setIsPublic ] = useState(true);
+    const [ loading, setLoading ] = useState(false);
+    const [ error, setError ] = useState('');
+
+    const investor = useParty();
+    const ledger = useLedger();
+
+    const handleSetupSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
```
Comment (Sarahb-da) [inline ui/src/components/Investor/Investor.tsx:106]: is this true for the issuer?

**Review by Sarahb-da — APPROVED** [review]
a few comments but looks good

File: ui/src/components/Investor/Investor.tsx (hunk 3 above)
Comment (alexmatson-da) [reply ui/src/components/Investor/Investor.tsx:106]: The issuer (and all the roles, actually) will have to accept an `IssuerInvitation` (or `{Role}Invitation`) contract. The only difference is the investor's invite contract is the only one that has any data that has to be supplied...

I've been working on the Exchange view so I've been thinking about this, and maybe for the other roles, we can just open a `useStreamFetchByKey` to listen for the invitation, and auto-exercise the choice in the frontend. If more data is added at this stage later, we can extract it out into a form then

**Review by dimitri-da — APPROVED** [review]
👍

**Pull request merged**